src/domain/hooks/hooks.repository.spec.ts

```diff
@@ -195,12 +200,42 @@ describe('HooksRepository (Unit)', () => {
     expect(mockSafeRepository.clearTransfers).not.toHaveBeenCalled();
     expect(mockSafeRepository.clearIncomingTransfers).not.toHaveBeenCalled();
   });
+
+  it('should store the unsupported chain events and log them after UNSUPPORTED_EVENTS_LOG_INTERVAL', async () => {
+    await hooksRepository.onModuleInit();
+    const chain = chainBuilder().build();
+    const event = incomingTokenEventBuilder()
+      .with('chainId', chain.chainId)
+      .build();
+    mockChainsRepository.isSupportedChain.mockResolvedValue(false);
+
+    await hooksRepository.onEvent(event);
+    await hooksRepository.onEvent(event);
+    await hooksRepository.onEvent(event);
+    await hooksRepository.onEvent(event);
+
+    jest.advanceTimersByTime(
+      HooksRepository.UNSUPPORTED_EVENTS_LOG_INTERVAL - 1,
+    );
+    expect(mockLoggingService.warn).not.toHaveBeenCalled();
+    jest.advanceTimersByTime(
+      HooksRepository.UNSUPPORTED_EVENTS_LOG_INTERVAL + 1,
+    );
+    expect(mockLoggingService.warn).toHaveBeenCalledTimes(1);
+    expect(mockLoggingService.warn).toHaveBeenCalledWith({
+      type: 'unsupported_chain_event',
+      chainId: chain.chainId,
+      count: 4,
+    });
+  });
 });
 
 describe('HooksRepositoryWithNotifications (Unit)', () => {
   let hooksRepositoryWithNotifications: HooksRepositoryWithNotifications;
 
   beforeEach(() => {
+    jest.useFakeTimers();
```

Comment:
Should we enable `fakeTimers` once at the beginning of the test suite using `beforeAll` and disable it again at the end by using `afterAll`?

Reply:
Moved in aea854488ca98389b1d3878f84dad1eadcadb337.